### PR TITLE
rdbc-663

### PR DESCRIPTION
**Pull request by alxsabo**
https://issues.hibernatingrhinos.com/issue/RDBC-663

### CODE REVIEW

File: src/Documents/Session/Loaders/QueryIncludeBuilder.php [hunk 1]
```diff
@@ -56,58 +60,52 @@ public function includeDocuments(?string $path): QueryIncludeBuilder
...
-//
-//    public function includeAllTimeSeries(TimeSeriesRangeType type, int count): QueryIncludeBuilder {
-//        _includeTimeSeriesByRangeTypeAndCount("", Constants.TimeSeries.ALL, type, count);
-//        return this;
-//    }
+    public function includeTimeSeriesRangeType(null | string | array $names, TimeSeriesRangeType $type, TimeValue | int $timeOrCount): QueryIncludeBuilder
+    {
+        if (is_string($names)) {
+            if (is_int($timeOrCount)) {
+//                $this->_includeTimeSeriesByRangeTypeAndCount("", $names, $type, $timeOrCount);
```
Comment (ml054) [inline src/Documents/Session/Loaders/QueryIncludeBuilder.php:85]: what about that?

File: src/Documents/Session/TimeSeries/TimeSeriesValuesHelper.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+use RavenDB\Exceptions\IllegalStateException;
+use RavenDB\Exceptions\RavenException;
+use ReflectionAttribute;
+use ReflectionClass;
+use ReflectionObject;
+
+class TimeSeriesValuesHelper
+{
+//    private static final ConcurrentMap<Class< ? ><!--, SortedMap<Byte, Tuple<Field, String>>> _cache = new ConcurrentHashMap<>();-->
+    private static array $cash = [];
```
Comment (ml054) [inline src/Documents/Session/TimeSeries/TimeSeriesValuesHelper.php]: cache? ;-)

**Comment by ml054** [comment]
@alxsabo Could you check failing test + fix (if needed my 2 remarks) before merge?

File: src/Documents/Session/Loaders/QueryIncludeBuilder.php (hunk 1 above)
Comment (alxsabo) [reply src/Documents/Session/Loaders/QueryIncludeBuilder.php:85]: none of the test required it yet, it just have to be uncommented. base method is implemented

File: src/Documents/Session/TimeSeries/TimeSeriesValuesHelper.php (hunk 2 above)
Comment (alxsabo) [reply src/Documents/Session/TimeSeries/TimeSeriesValuesHelper.php]: cash is better :D

File: src/Documents/Session/Loaders/QueryIncludeBuilder.php (hunk 1 above)
Comment (ml054) [reply src/Documents/Session/Loaders/QueryIncludeBuilder.php:85]: ok sure. :)

**Review by ml054 — APPROVED** [review]

**Comment by ml054** [comment]
Great let's wait for GH actions and we can merge. :)

**Pull request merged**